src/solver/opamCudf.ml

```diff
@@ -864,10 +852,10 @@ let extract_explanations packages cudfnv2opam reasons : explanation list =
           let formula =
             OpamFormula.formula_of_version_set all_versions versions
           in
-          OpamFormula.to_string (Atom (name, formula)))
+          Atom (name, formula))
         nvs
     in
-    String.concat ", " (OpamPackage.Name.Map.values strs)
+    `Formula (false, OpamFormula.ors (OpamPackage.Name.Map.values strs))
   in
   let cs_to_string ?(hl_last=true) cs =
```

Comment:
same here, `construct_cs` ?

Reply:
the return type of this function is still a string. Also `cs` is different from `cst/construct` and is a different concept. I've improved the naming difference in 3b9d8032e